src/AutoRest.CSharp/Common/Output/Models/Types/SchemaObjectType.cs

```diff
@@ -631,5 +645,21 @@ protected override XmlObjectSerialization EnsureXmlSerialization()
         {
             return _serializationBuilder.BuildXmlObjectSerialization(ObjectSchema, this);
         }
+
+        private ObjectType? BuildDefaultDerviedType()
+        {
+            if (_hasCalculatedDefaultDerivedType)
+                return _defaultDerivedType;
+
+            _hasCalculatedDefaultDerivedType = true;
+            if (_context.BaseLibrary is null)
+                return null;
+
+            var defaultDerivedSchema = ObjectSchema.GetDefaultDerivedSchema();
+            if (defaultDerivedSchema is null)
+                return null;
+
+            return _context.BaseLibrary.FindTypeProviderForSchema(defaultDerivedSchema) as ObjectType;
```

Comment:
Looks like this is the only place to use this as a TypeProvider. We actually do not need to introduce this new abstract method, we could use the [`TryCast<ObjectType>` method](https://github.com/Azure/autorest.csharp/blob/943b721046b5a5cf94ee80e2c8e3bc3b2176ae9e/src/AutoRest.CSharp/Mgmt/Decorator/TypeExtensions.cs#L65) on the CSharpType returned by `FindTypeForSchema`.
We maybe we should throw when the type is not a `ObjectType` (well in the current implementation it is always a ObjectType)

Reply:
I actually thing I would want to deprecate the CSharpType version of this abstract method since TypeProvider is a superset of the info (even though CSharpType does have a two way reference).

For now I will leave this as there are other areas of consolidation that will be coming in the future.